packages/auth/amplify_auth_cognito_dart/lib/src/flows/hosted_ui/hosted_ui_platform.dart

```diff
@@ -95,27 +95,30 @@ abstract class HostedUiPlatform implements Closeable {
   @protected
   @visibleForTesting
   @nonVirtual
-  Uri getSignInUri({
+  Future<Uri> getSignInUri({
     Uri? redirectUri,
     AuthProvider? provider,
-  }) {
+  }) async {
     final state = generateState();
     final codeVerifier = createCodeVerifier();
     final nonce = generateState();
 
-    _secureStorage
-      ..write(
-        key: _keys[HostedUiKey.state],
-        value: state,
-      )
-      ..write(
-        key: _keys[HostedUiKey.codeVerifier],
-        value: codeVerifier,
-      )
-      ..write(
-        key: _keys[HostedUiKey.nonce],
-        value: nonce,
-      );
+    await Future.wait<void>(
```

Comment:
Was this a bug that the tests uncovered?

Reply:
No but maybe I should verify my assumptions that it's needed